kolibri/core/assets/src/views/k-grid/index.vue

```diff
@@ -1,19 +1,104 @@
 <template>
 
-  <div class="pure-g">
-    <slot></slot>
+  <div>
+    <div
+      class="pure-g"
+      :style="{ marginLeft: marginOffset, marginRight: marginOffset }"
+    >
+      <slot></slot>
+    </div>
+    <overlay
+      v-if="debug"
+      :cols="actualNumCols"
+      :gutterWidth="windowGutter"
+    />
   </div>
 
 </template>
 
 
 <script>
 
+  import responsiveWindow from 'kolibri.coreVue.mixins.responsiveWindow';
+  import logger from 'kolibri.lib.logging';
+  import overlay from './overlay';
+
+  const logging = logger.getLogger(__filename)
+
+
+  /**
+   * Grid layouts
+   */
   export default {
     name: 'kGrid',
+    components: { overlay },
+    mixins: [responsiveWindow],
+    props: {
+      /**
+       * Set a fixed number of columns, bypassing default responsive behavior.
+       * This can be useful for nesting grids.
+       */
+      cols: {
+        type: [Number, String],
+        required: false,
+        validator(value) {
+          if (value < 2 || value > 12) {
+            logging.error(`Number of columns (${value}) is not between 2 and 12`);
+            return false;
+          }
+          return true;
+        },
+      },
+      debug: {
+        type: Boolean,
+        default: false,
+      },
```

Comment:
Pass in a `debug` attribute to help visualize your grid! For example:

```html
<k-grid debug> ... </k-grid>
```

It's doesn't always work perfectly, but it's helpful anyway

![image](https://user-images.githubusercontent.com/2367265/42965324-811a6748-8b4e-11e8-8224-e4d1fc0dfd63.png)